overlord/fdestate/backend/reseal.go

```diff
@@ -70,6 +70,17 @@ type SealingParameters struct {
 	TpmPCRProfile []byte
 }
 
+// EncryptedDisk gives information on the role, path and path to extra
+// legacy keys.
+type EncryptedDisk interface {
+	// GetRole gives the container role of the disk. See KeyslotRoleInfo.Parameters.
+	GetRole() string
```

Comment:
given that `role` is seriously abused already in fdestate & secboot, maybe GetContainerRole() so that it's clear what is returned here?

Reply:
I forgot, but according to golang naming, it would be better if we could not have all these "Get" prefixes here

Reply:
Done.